### DISCUSSION TITLE
Script and HTML imports

### DISCUSSION THREAD

**Discussion by filipw01** [crxjs/chrome-extension-tools#269] [Q&A] [answered]
@jacksteamdev 
Do script and HTML imports work in 4.0.1-18?
I'm getting the following error for script imports `import scriptFileName from './testa?script'`
```
  crx:dynamic serve:resolveId : source ./testa?script +113ms
  crx:dynamic serve:resolveId : chunk id /Users/filip/projects/chilipiper-frontend/instant-booker-extension/src/testa.ts +3ms
  crx:dynamic serve:resolveId : id /crx-dynamic/src/testa.ts +1ms
  crx:dynamic serve:resolveId : fileName src/testa.ts +0ms
✘ [ERROR] [plugin vite:dep-scan] ENOENT: no such file or directory, open '/crx-dynamic/src/testa.ts'
```
And errors for `import htmlFileName from './testa.html'` like
```
Failed to parse source for import analysis because the content contains invalid JS syntax. You may need to install appropriate plugins to handle the .html file format.
5:21:06 PM [vite] Internal server error: Failed to parse source for import analysis because the content contains invalid JS syntax. You may need to install appropriate plugins to handle the .html file format.
```

**Comment by jacksteamdev** [comment, marked as answer]
@filipw01 Sorry, there was a regression for script imports in v4.0.1-18. I've got a working fix, but I need to cover some edge cases before release. I'll let you know when the next version lands.

Script imports will work as before, HTML pages not declared in the manifest will work the same as a [Vite Multi-page app](https://vitejs.dev/guide/build.html#multi-page-app).

**Reply by filipw01** [reply]
Thank you for the quick answer 😄
